axelor-sale/src/main/resources/views/SaleOrder.xml

```diff
@@ -1520,6 +1550,34 @@
     <action name="action-sale-order-method-finalize-quotation"/>
   </action-group>
 
+  <action-group name="action-sale-order-group-new-version-btn">
+    <action name="save"/>
+    <action name="action-sale-order-validate-check-empty-sale-order-on-new-version"/>
+    <action name="action-sale-order-method-new-version"/>
+    <action name="action-sale-order-attrs-set-previous-version-number"/>
+    <action name="action-sale-order-method-get-version-date-time"/>
+    <action name="save"/>
+  </action-group>
+
+  <action-group name="action-sale-order-group-previous-version-number-on-change">
+    <action name="action-sale-order-attrs-previous-version-number-on-change"/>
+    <action name="action-sale-order-method-get-version-date-time"/>
+  </action-group>
+
+  <action-group name="action-sale-order-group-last-version-btn">
+    <action name="action-sale-order-attrs-set-previous-version-number"/>
+    <action name="action-sale-order-attrs-previous-version-number-on-change"/>
+    <action name="action-sale-order-method-get-version-date-time"/>
+  </action-group>
+
+  <action-group name="action-sale-order-group-version-recovery-btn">
+    <action name="action-sale-order-method-version-recovery-btn"/>
+    <action name="action-sale-order-attrs-set-previous-version-number"/>
+    <action name="action-sale-order-method-get-version-date-time"/>
+    <action name="action-sale-order-line-change-group"/>
+    <action name="save"/>
```

Comment:
If you have a save on clicking on a button, the `save` should be the first action, then the action-method should update the records in database and call `response.setReload(true)`